store/mockstore/mocktikv/cop_handler_dag.go

```diff
@@ -100,7 +146,17 @@ func (h *rpcHandler) buildDAGExecutor(req *coprocessor.Request) (*dagContext, ex
 		return nil, nil, nil, errors.Trace(err)
 	}
 	sc := flagsToStatementContext(dagReq.Flags)
-	sc.TimeZone = time.FixedZone("UTC", int(dagReq.TimeZoneOffset))
+
+	// retrieving timezone by name first. When name is set, it means we need
+	// consider daylight saving time. If it is not, we can use offset.
+	if dagReq.TimeZoneName != "" {
+		if sc.TimeZone, err = LocCache.getLoc(dagReq.TimeZoneName); err != nil {
+			return nil, nil, nil, errors.Trace(err)
+		}
+		dagReq.TimeZoneName = sc.TimeZone.String()
```

Comment:
Why set `dagReq.TimeZoneName `?

Reply:
We set `dagReq.TimeZoneOffset` here, so do `dagReq.TimeZoneName`.